service/src/main/kotlin/fi/espoo/evaka/reports/OccupancyReport.kt

```diff
@@ -470,41 +590,41 @@ WHERE sn.placement_id = ANY(:placementIds)
         return assistanceCoefficient * serviceNeedCoefficient
     }
 
-    val placementsAndPlans = (placements + placementPlans).groupBy { it.unitId }
+    val placementsAndPlans = (placements + placementPlans).groupBy { it.groupingId }
 
-    return caretakers
-        .map { (unit, values) ->
+    return caretakerCounts
+        .map { (key, values) ->
             val occupancies = values
-                .associate { (_, date, caretakers) ->
-                    val unitPlacementsOnDate = (placementsAndPlans[unit.unitId] ?: listOf())
-                        .filter { it.period.includes(date) }
-                        .filterNot { absences[it.childId]?.any { absence -> absence.date == date } ?: false }
+                .associate { caretakers ->
+                    val unitPlacementsOnDate = (placementsAndPlans[key.groupingId] ?: listOf())
+                        .filter { it.period.includes(caretakers.date) }
+                        .filterNot { absences[it.childId]?.any { absence -> absence.date == caretakers.date } ?: false }
 
                     val sum = unitPlacementsOnDate
                         .groupBy { it.childId }
-                        .mapNotNull { (_, placements) -> placements.map { getCoefficient(date, it) }.maxOrNull() }
+                        .mapNotNull { (_, placements) ->
+                            placements.map { getCoefficient(caretakers.date, it) }.maxOrNull()
+                        }
                         .fold(BigDecimal.ZERO) { sum, value -> sum + value }
 
                     val percentage =
-                        if (caretakers.compareTo(BigDecimal.ZERO) == 0) null
+                        if (caretakers.caretakerCount.compareTo(BigDecimal.ZERO) == 0) null
                         else sum
-                            .divide(caretakers * BigDecimal(7), 10, RoundingMode.HALF_UP)
+                            .divide(caretakers.caretakerCount * BigDecimal(7), 10, RoundingMode.HALF_UP)
```

Comment:
Not sure if using BigDecimal api is a bit of an overkill here, but the readability isn't too bad so maybe thats good 🤷‍♂️

Reply:
I'm not completely sure what you mean, but I think using kotlin's `/` operator with `BigDecimal` doesn't make it very clear what the end result is so I dislike using it.

Reply:
yeah i mean i was wondering whether a normal Double would be enough here since this is not about money and we dont care that much about rounding errors